src/math-history.c

```diff
@@ -50,33 +50,54 @@ scroll_bottom_cb(MathHistory *history, gpointer data)
                                           - gtk_adjustment_get_page_size (adjustment));
 }
 
-static gboolean
-check_history(MathHistory *history, char *equation)
-{ /* Checks if the last inserted calculation is the same as the current calculation to be inserted in history */
-    if (history->priv->items_count >= 1 && g_strcmp0(equation, history->priv->prev_equation)==0)
-        return TRUE; /* returns true if last entered equation is the same as the current equation */
-    else
-        return FALSE;
-}
-
 void
 math_history_insert_entry (MathHistory *history, char *equation, MPNumber *answer, int number_base)
-{   /* Inserts a new entry into the history listbox */
-    history->priv->entry = math_history_entry_new(history->priv->equation);
-    gboolean check = check_history (history, equation);
-    history->priv->prev_equation = g_strdup(equation);
-    if (!check)
+{
+    MathHistoryEntry *entry = math_history_entry_new(history->priv->equation);
+
+    MpSerializer *serializer_four = mp_serializer_new(MP_DISPLAY_FORMAT_AUTOMATIC, number_base, 4);
+    MpSerializer *serializer_nine = mp_serializer_new(MP_DISPLAY_FORMAT_AUTOMATIC, number_base, 9);
+
+    gchar *answer_four_digits = mp_serializer_to_string(serializer_four, answer);
+    gchar *answer_nine_digits = mp_serializer_to_string(serializer_nine, answer);
+
+    if (g_strcmp0(history->priv->last_answer, answer_nine_digits) == 0 &&
+        g_strcmp0(history->priv->last_equation, equation) == 0 &&
+        history->priv->rows >= 1)
     {
-        math_history_entry_insert_entry(history->priv->entry, equation, answer, number_base);
-        if (history->priv->entry != NULL)	
-        {
-            gtk_list_box_insert(GTK_LIST_BOX(history->priv->listbox), GTK_WIDGET(history->priv->entry), -1);
-            gtk_widget_set_can_focus(GTK_WIDGET(history->priv->entry), FALSE);
-            gtk_widget_show_all(GTK_WIDGET(history->priv->entry));	
-            history->priv->items_count++;
-        }
+        g_free(answer_four_digits);
+        g_free(answer_nine_digits);
+        return;
     }
+
+    math_history_entry_insert_entry(entry, equation, answer_four_digits, answer_nine_digits);
+
+    gtk_list_box_insert(GTK_LIST_BOX(history->priv->listbox), GTK_WIDGET(entry), -1);
+    gtk_widget_set_can_focus(GTK_WIDGET(entry), false);
```

Comment:
FALSE instead false to make it more readable, please

Reply:
you are right